test/framework/interfaces.go

```diff
@@ -0,0 +1,58 @@
+/*
+Copyright 2019 The Kubernetes Authors.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package framework
+
+import (
+	"context"
+
+	"sigs.k8s.io/controller-runtime/pkg/client"
+)
+
+// ComponentGenerator is used to install components, generally any YAML bundle.
+type ComponentGenerator interface {
+	// GetName returns the name of the component.
+	GetName() string
+	// Components return the YAML bundle.
+	Components() ([]byte, error)
```

Comment:
We should call this what it is :P `Yaml()`, `Manifests()`, `Spec()` or something else

Reply:
The reason behind calling this components is that we call them components in all our release artifacts. For example:

* https://github.com/kubernetes-sigs/cluster-api/releases/download/v0.2.7/cluster-api-components.yaml
* https://github.com/kubernetes-sigs/cluster-api-provider-aws/releases/download/v0.4.4/infrastructure-components.yaml

I don't really care what it's called, but the consistency is nice here. What are the reasons behind renaming this to `Yaml()` or `Manifests()` from your perspective?

Reply:
Went with Manifests